app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/detail/DetailScreenUiTest.kt

```diff
@@ -0,0 +1,160 @@
+package co.nimblehq.compose.crypto.ui.screen.detail
+
+import androidx.activity.compose.setContent
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import co.nimblehq.compose.crypto.R
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinDetailUseCase
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinPricesUseCase
+import co.nimblehq.compose.crypto.extension.toFormattedString
+import co.nimblehq.compose.crypto.test.MockUtil.coinDetail
+import co.nimblehq.compose.crypto.test.MockUtil.coinPrices
+import co.nimblehq.compose.crypto.ui.BaseScreenTest
+import co.nimblehq.compose.crypto.ui.components.chartintervals.TimeIntervals
+import co.nimblehq.compose.crypto.ui.navigation.AppDestination
+import co.nimblehq.compose.crypto.ui.screens.MainActivity
+import co.nimblehq.compose.crypto.ui.screens.detail.*
+import co.nimblehq.compose.crypto.ui.screens.home.FIAT_CURRENCY
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import kotlin.math.abs
+
+@ExperimentalCoroutinesApi
+class DetailScreenUiTest : BaseScreenTest() {
+
+    @get:Rule
+    val composeAndroidTestRule = createAndroidComposeRule<MainActivity>()
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val mockGetCoinDetailUseCase: GetCoinDetailUseCase = mockk()
+    private val mockGetCoinPricesUseCase: GetCoinPricesUseCase = mockk()
+
+    private lateinit var detailViewModel: DetailViewModel
+
+    private var appDestination: AppDestination? = null
+
+    @Before
+    fun setUp() {
+        initDetailViewModel()
+        composeAndroidTestRule.activity.setContent {
+            DetailScreen(
+                coinId = "",
+                viewModel = detailViewModel,
+                navigator = { destination ->
+                    appDestination = destination
+                }
+            )
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_shows_loading() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flow {
+            delay(100L)
+            emit(coinDetail)
+        }
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flow {
+            delay(100L)
+            emit(coinPrices)
+        }
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(TestTagDetailLoading).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun when_navigating_to_detail_screen_it_renders_chart_interval_buttons_properly() {
+        every { mockGetCoinDetailUseCase.execute(any()) } returns flowOf(coinDetail)
+        every { mockGetCoinPricesUseCase.execute(any()) } returns flowOf(coinPrices)
```

Comment:
We can mock them as default at the `setUp` block, and override a response when needed to reduce duplicated mocking in the following test cases (4 times).